datastore/src/main/java/io/spine/server/storage/datastore/config/StorageConfiguration.java

```diff
@@ -0,0 +1,238 @@
+         *
+         * @param context
+         *         the specification of the Bounded Context
+         * @return this instance of this {@code Builder}, for call chaining
+         */
+        public Builder<I, R> withContext(ContextSpec context) {
+            this.context = checkNotNull(context);
+            return this;
+        }
+
+        /**
+         * Sets the Datastore wrapper to use by the storage.
+         *
+         * <p>This parameter is mandatory.
+         *
+         * <p>Overrides the previous value, if set.
+         *
+         * @param datastore
+         *         the Datastore wrapper
+         * @return this instance of this {@code Builder}, for call chaining
+         */
+        public Builder<I, R> withDatastore(DatastoreWrapper datastore) {
+            this.datastore = checkNotNull(datastore);
+            return this;
+        }
+
+        /**
+         * Sets the record specification for the configured storage.
+         *
+         * <p>This parameter is mandatory.
+         *
+         * <p>Overrides the previous value, if set.
+         *
+         * @param recordSpec
+         *         the record specification
+         * @return this instance of this {@code Builder}, for call chaining
+         */
+        public Builder<I, R> withRecordSpec(DsEntitySpec<I, R> recordSpec) {
+            this.recordSpec = checkNotNull(recordSpec);
+            return this;
+        }
+
+        /**
+         * Sets the column type mapping for the configured storage
+         *
+         * <p>This parameter is mandatory.
+         *
+         * <p>Overrides the previous value, if set.
+         *
+         * @param columnMapping
+         *         the column type mapping
+         * @return this instance of this {@code Builder}, for call chaining
+         */
+        public Builder<I, R> withMapping(ColumnMapping<Value<?>> columnMapping) {
+            this.columnMapping = checkNotNull(columnMapping);
+            return this;
+        }
+
+        /**
+         * Sets the transactional setting for the configured storage.
+         *
+         * <p>This parameter is optional. By default, transactional behavior is set
+         * to {@linkplain TxSetting#disabled() disabled}.
+         *
+         * <p>Overrides the previous value, if set.
+         *
+         * @param txSetting
+         *         the setting of transactional behavior
+         * @return this instance of this {@code Builder}, for call chaining
+         */
+        public Builder<I, R> withTxSetting(TxSetting txSetting) {
+            this.txSetting = checkNotNull(txSetting);
+            return this;
+        }
+
+        public StorageConfiguration<I, R> build() {
+            checkNotNull(context, "`ContextSpec` must be set.");
+            checkNotNull(datastore, "`DatastoreWrapper` must be set.");
+            checkNotNull(recordSpec, "`DsRecordSpec` must be set.");
+            checkNotNull(columnMapping, "`ColumnMapping` must be set.");
```

Comment:
This block of checks suggests for having a helper function which takes a simple class name of a passed field.